### PR TITLE
Cherry-picks for 2.10.23-RC.6

### PR DESCRIPTION
**Pull request by neilalexander**
Includes the following:

- https://github.com/nats-io/nats-server/pull/6169
- https://github.com/nats-io/nats-server/pull/6165
- https://github.com/nats-io/nats-server/pull/6173
- https://github.com/nats-io/nats-server/pull/6175
- https://github.com/nats-io/nats-server/pull/6176
- https://github.com/nats-io/nats-server/pull/6178

Signed-off-by: Neil Twigg <neil@nats.io>

### CODE REVIEW

**Review by derekcollison — APPROVED** [review]
LGTM

File: go.mod [hunk 1]
```diff
@@ -1,8 +1,11 @@
 module github.com/nats-io/nats-server/v2
 
-go 1.21.0
+go 1.22
+
+toolchain go1.23.0
```
Comment (wallyqs) [inline go.mod:5]:
```suggestion
toolchain go1.22.8```

**Comment by neilalexander** [comment]
Have fixed the Go version bump in `go.mod`.

File: go.mod [hunk 2]
```diff
@@ -1,8 +1,9 @@
 module github.com/nats-io/nats-server/v2
 
-go 1.21.0
+go 1.21
```
Comment (wallyqs) [inline go.mod:3]:
```suggestion
go 1.21.0
```

**Review by wallyqs — APPROVED** [review]
LGTM

**Pull request merged**